apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/trace/AbstractTracingSpan.java

```diff
@@ -349,4 +349,10 @@ public AbstractSpan start(long startTime) {
         isAsyncStopped = true;
         return this;
     }
+
+    @Override
+    public AbstractSpan setOperationNameInAsync(String operationName) {
+        return setOperationName(operationName);
```

Comment:
You should check the span is in async mode. Meanwhile, you should make the `setOperationName` disable when in async mode. Right?

Reply:
Also, new API should be introduced in `Java-Plugin-Development-Guide.md` doc